### PR TITLE
Update checked auth

### PR DESCRIPTION
**Pull request by tranlehaiquan**
Just check if user login one, delete axios header Authorization if user logout. So they can't access /settings or /editor.
We will do sync tab later.
#35

### CODE REVIEW

**Comment by tecfu** [comment]
Hi @tranlehaiquan, I found this bug before I found your previously submitted pull request for it. 

[Here's](https://github.com/gothinkster/vue-realworld-example-app/pull/45) how I decided to fix the same issue, because I wanted to keep [router.beforeEach()](https://github.com/gothinkster/vue-realworld-exampleapp/blob/a56e4c049b077e5b886f83f99e08a0c3e2c16512/src/main.js#L21-L27).

I'm curious about a couple things you did here though and wanted to ask you about them. 

-

File: src/main.js [hunk 1]
```diff
@@ -17,20 +17,14 @@ Vue.filter('error', ErrorFilter)
...
-
-/* eslint-disable no-new */
-new Vue({
-  el: '#app',
-  router,
-  store,
-  template: '<App/>',
-  components: { App }
-})
+store.dispatch(CHECK_AUTH)
```
Comment (tecfu) [inline src/main.js]: This runs only once on init, correct?

File: src/store/auth.module.js [hunk 2]
```diff
@@ -49,19 +49,25 @@ const actions = {
...
+        ApiService
+          .get('user')
+          .then(({data}) => {
+            context.commit(SET_AUTH, data.user)
+          })
+          .catch(({response}) => {
+            context.commit(SET_ERROR, response.data.errors)
+          })
+          .finally(() => {
+            resolve()
```
Comment (tecfu) [inline src/store/auth.module.js]: Why not pass the user data to resolve within the 

```
.then
```
and
```
.catch
```
calls? 

This way we can use it with an auth library like [CASL](https://www.npmjs.com/package/@casl/vue).

File: src/main.js (hunk 1 above)
Comment (tranlehaiquan) [reply src/main.js]: Yes, it just run once.
(If something wrong in here please tell me 👍 )

File: src/store/auth.module.js (hunk 2 above)
Comment (tranlehaiquan) [reply src/store/auth.module.js]: I think at the moment I wrote this, I want be sure that, this always be resolve so I use finally.
And about CASL I use vue-router meta for check if that page has required authentication or not, so I decide direct to login page or not. If you have time please check my little aside project: (I think this project not work anymore :D)
https://github.com/tranlehaiquan/conduit-vue

**Pull request closed without merging**